### ISSUE TITLE
The assignment behavior of DGLGraph.dstdata is wrong in some cases

### ISSUE THREAD

**Issue by yfismine** [dmlc/dgl#7452]
## To Reproduce

```python
import dgl
from openhgnn.dataset.gtn_dataset import ACM4GTNDataset

graph = dgl.to_block(ACM4GTNDataset()[0])
print(graph)
print(graph.ndata["label"])
graph.dstdata["label_copy"] = graph.dstdata["label"]
print(graph.dstdata["label_copy"])
print(graph.srcdata["label_copy"])
graph.dstnodes["paper"].data["label_copy"] = next(iter(graph.dstdata["label"].values()))
print(graph.dstdata["label_copy"])

```
<img width="982" alt="image" src="https://github.com/dmlc/dgl/assets/53209596/193fe447-5634-4a8b-8085-333acfbc4c4b">

## Expected behavior
You can see that the attribute directly assigned to dstdata can't be found by querying dstdata. On the contrary, it is amazing to find this attribute in srcdata. I checked the code of HeteroNodeDataView and found that, The ntid used by self._graph._set_n_repr is through self._graph.get_ntype_id(ntype), but in get_ntype_id, ntid = self. _ srctypes _ invmap.get (ntype, Self. _ dsttypes _ invmap.get (ntype, none)), that is to say, the obtained ntid gives priority to srctypes, so this bug will be triggered if ntype appears in both the source node and the destination node.

## Environment
 - DGL Version : 2.2.1
 - Backend Library & Version : Pytorch 2.3.0
 - OS : Linux
 - How you installed DGL : conda
 - Python version: 3.12.3
 - CUDA/cuDNN version (if applicable): 11.8
 - GPU models and configuration : nvidia 3090

**Comment by frozenbugs** [comment]
What's your use case of using to_block()?

**Comment by yfismine** [comment]
> What's your use case of using to_block()?

DGLGraph
![image](https://github.com/dmlc/dgl/assets/53209596/a45432c4-b993-4a24-95d6-8ae5c002f223)

**Comment by frozenbugs** [comment]
to_block is to create a bipartite-structured block for message passing. It is more an internal util method than public api of dgl, we expose it just in case users want to experiment with it. Can you be more specific about why you need to use to_block to create a bipartite-structured block? Can DGLGraph fit your need?

**Comment by yfismine** [comment]
> to_block is to create a bipartite-structured block for message passing. It is more an internal util method than public api of dgl, we expose it just in case users want to experiment with it. Can you be more specific about why you need to use to_block to create a bipartite-structured block? Can DGLGraph fit your need?

It is found that the assignment behavior here is normal if to_block is not used. I encountered this problem when I used dist dgl for neighbor sampling and found that the generated block has no attributes. I need to pull data through DistGraph in a way like "batch _ labels = g. ndata ["labels "] [seeds]. long (). To (device)". I hope to unify the training codes of stand-alone and distributed. I re-implement the sampler_block of neighborSampler and assign attributes to the block in it.
